### PR TITLE
Add odhDeployment Custom Resource Definition

### LINKED ISSUE #164: Add DataScienceCluster CRD and define DataScienceClusterSpec
This issue tracks design for the DataScienceCluster CRD.
- [x] #231

### PR DESCRIPTION
**Pull request by VaishnaviHire**
Reference Architecture: https://docs.google.com/document/d/1QosVwvjWF0o-7HRE4zUEnEQPdt3uDPnNS0ucL156y40/edit#heading=h.2nur9rvqde54 

Closes #164 
This PR just involves defining the proposed `OdhDeployment` CRD as explained in the architecture proposal. It doesn't include controller implementation.

Following is the list of new fields introduced as part of the CRD:
- `enabled` : Allows a user to install/uninstall a single component
- `reconcile` : If set to false, allows user to update component resources
-  `version` : Current version of the manifests. The user can update this field to update the manifests to a new version.

Example [CR](https://github.com/opendatahub-io/opendatahub-operator/blob/422f4bc07989e26931dc74079e6074eac2a94a48/config/samples/_v1alpha1_odhdeployment.yaml)

### CODE REVIEW

File: config/crd/bases/opendatahub.io_odhdeployments.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: OdhDeploymentSpec defines the desired state of OdhDeployment
+            properties:
+              components:
+                description: 'INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
```
Comment (anishasthana) [inline config/crd/bases/opendatahub.io_odhdeployments.yaml:39]: This doesn't seem right

**Review by anishasthana — COMMENTED** [review]
Just have the one comment

**Comment by shalberd** [comment]
Regarding repos.name.uri, will the download functionality code be in the same place as previously with the operator?

**Pull request closed without merging**